### ISSUE TITLE
Scrum spike: Research prompt engineering techniques

### ISSUE THREAD

**Issue by tubamos** [amosproj/amos2024ss06-health-ai-framework#160]

**Comment by preetvadaliya** [comment]
@tubamos , 

Just an idea for research tasks can we create a notebook and submit them in sprint-dilevriables so that can be stored there and in next sprint we dont need to remove or fix them.

**Comment by tubamos** [comment]
@preetvadaliya yes, let's do this. We actually discussed this option in the past but we did not conclude on whether we do it that way or not because we wanted to check what options exist in github to make such tasks trackable for formal reasons. 

The option to commit any documents in the project folder as you suggest has been one option. I am fine with that if the devs agree.

**Comment by manikg08** [comment]
Different Prompt engineering techniques have been added to the code and have been pushed into the main branch. The current problem with prompt engineering is that all the scraped data is in a different format like each scraper has a different JSON structure due to which chunking is quite problematic. We need to create different classes for chunking different scraped data into the Vector Database.

**Comment by tubamos** [comment]
@manikg08 Let us disucss this during the sprint meeting today.
